examples/qml_features/rust/src/custom_base_class.rs

```diff
@@ -31,6 +31,15 @@ pub mod qobject {
         type QVector_i32 = cxx_qt_lib::QVector<i32>;
     }
 
+    #[qenum(CustomBaseClass)]
```

Comment:
would be good to show this in use in QML and also to have a unit test in the QML tests (the test being the most important)

You could be able to try doing `model.data(index, CustomBaseClass.Value);` or something somewhere in JavaScript to demo that.

Reply:
I'll try adding a test to the QML tests.

Reply:
Done.

Reply:
Hmm... it kinda bugs me that `CustomBaseClass` isn't in scope here because the type alias is declared in the `extern "RustQt"` block below, making this feel a bit magical. We could move this into the `extern "RustQt"` block, but that would break from upstream CXX where shared enums are outside the `extern` blocks... thoughts? I don't have a strong opinion either way.

Reply:
Isn't this the same as CXX though? you can define a shared enum or struct and use it in another block below or above ? The `type T` is enough to make that `T` in scope for the whole bridge?

And the enum here was originally trying to emulate the shared enums of CXX but for Qt.

The Q_ENUM_NS variant (coming in a later PR?) would just be `#[qenum]` however :-)